<a href="https://colab.research.google.com/github/GreatUsername63/Obtencion-de-estadisticas-descriptivas/blob/main/Obtencion_de_estadisticas_descriptivas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [12]:
from google.colab import files

#Subir el archivo Iris.csv
uploaded = files.upload()

for fn in uploaded.keys():
  print('User uploaded file "{name}" with length {length} bytes'.format(
      name=fn, length=len(uploaded[fn])))

Saving iris.csv to iris.csv
User uploaded file "iris.csv" with length 3867 bytes


In [2]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv('iris.csv')
df.head()

Sepal.Length  Sepal.Width  Petal.Length  Petal.Width Species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Sepal.Length  150 non-null    float64
 1   Sepal.Width   150 non-null    float64
 2   Petal.Length  150 non-null    float64
 3   Petal.Width   150 non-null    float64
 4   Species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [5]:
df.shape

(150, 5)

In [11]:
print('Rango')
print('Sepal Length: ', df['Sepal.Length'].max() - df['Sepal.Length'].min())
print('Sepal Width: ', df['Sepal.Width'].max() - df['Sepal.Width'].min())
print('Petal Length: ', df['Petal.Length'].max() - df['Petal.Length'].min())
print('Petal Width: ', df['Petal.Width'].max() - df['Petal.Width'].min())

Rango
Sepal Length:  3.6000000000000005
Sepal Width:  2.4000000000000004
Petal Length:  5.9
Petal Width:  2.4


In [8]:
print('Media')
print('Sepal Length: ', df['Sepal.Length'].mean())
print('Sepal Width: ', df['Sepal.Width'].mean())
print('Petal Length: ', df['Petal.Length'].mean())
print('Petal Width: ', df['Petal.Width'].mean())

Media
Sepal Length:  5.843333333333334
Sepal Width:  3.0573333333333337
Petal Length:  3.7580000000000005
Petal Width:  1.1993333333333336


In [9]:
print('Mediana')
print('Sepal Length: ', df['Sepal.Length'].median())
print('Sepal Width: ', df['Sepal.Width'].median())
print('Petal Length: ', df['Petal.Length'].median())
print('Petal Width: ', df['Petal.Width'].median())

Mediana
Sepal Length:  5.8
Sepal Width:  3.0
Petal Length:  4.35
Petal Width:  1.3


In [10]:
print('Desviacion estandar:')
print('Sepal Length: ', df['Sepal.Length'].std())
print('Sepal Width: ', df['Sepal.Width'].std())
print('Petal Length: ', df['Petal.Length'].std())
print('Petal Width: ', df['Petal.Width'].std())

Desviacion estandar:
Sepal Length:  0.828066127977863
Sepal Width:  0.4358662849366982
Petal Length:  1.7652982332594662
Petal Width:  0.7622376689603465


Conclusion:
Los datos numericos sobre las dimensiones de los petalos y sepalos de estas plantas Iris, nos da a entender que la planta promedio tiene un sepalo de 5.84x3.06cm y petalos de 3.75x1.20cm. Una planta Iris puede tener una diferencia de petalos de hasta 5.9cm de longitud y 2.4cm de anchura y una diferencia de 3.6 de longitud y 2.4cm de anchura en los sepalos. La planta mediana en estas estadisticas es una Iris con sepalos de 5.8x3.0cm y 4.35x1.3cm en los petalos. Por ultimo, la desviación estandar nos indica que el valor que mas varia entre flor y flor es el de la longitud de los petalos mientras que el que menos varia es el de la anchura del sepalo.